src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentInvoker.cs

```diff
@@ -179,11 +157,13 @@ public IViewComponentResult InvokeSyncCore(MethodInfo method, ViewComponentConte
             {
                 _diagnosticSource.BeforeViewComponent(context, component);
                 _logger.ViewComponentExecuting(context);
-
+                var indexedArguments = ControllerActionExecutor.PrepareArguments(
```

Comment:
In the event an argument does not have a value?

Component.InvokeAsync("Foo", new { a = "some value" }) -->  InvokeAsync(object a, object b)